pkg/archive/archive_zip.go

```diff
@@ -145,3 +166,17 @@ func (e gozipArchiveFactory) OpenBytes(data []byte, password string) (Archive, e
 	}
 	return NewGoZIPArchive(r, func() error { return nil }), nil
 }
+
+type ReaderAtCloser interface {
+	io.Closer
+	io.ReaderAt
+}
+
+func (e gozipArchiveFactory) OpenReader(reader ReaderAtCloser, size int64, password string) (Archive, error) {
+	// Go's built-in zip reader doesn't support passwords. Maybe look into something like https://github.com/mholt/archiver
```

Comment:
I'd like to leave it in the interface for:
- Staying similar to the other toolkits
- Support for other formats, possibly DRMed
- Someone may want to write an implementation of ArchiveFactory, and the password option should be left there if they need it
I have altered gozipFactory to refuse to extract if a password is provided, to prevent confusion/misunderstanding

Reply:
Sounds good.